app/assets/javascripts/discourse/widgets/dropdown.js

```diff
@@ -0,0 +1,266 @@
+import { createWidget } from "discourse/widgets/widget";
+import hbs from "discourse/widgets/hbs-compiler";
+
+/*
+
+  dropdown widget
+
+  Usage
+  -----
+
+  {{attach
+    widget="dropdown"
+    attrs=(hash
+      id=id
+      label=label
+      content=content
+      onChange=onChange
+      options=(hash)
+    )
+  }}
+
+  Mandatory attributes:
+
+    - id: must be unique in the application
+
+    - label: an i18n key to be translated and displayed on the header
+
+    - translatedLabel: an already translated label to display on the header
+
+    - onChange: action called when a click happens on a row, content[rowIndex] will be passed as params
+
+  Optional attributes:
+
+    - class: adds css class to the dropdown
+    - content: list of items to display, if undefined or empty dropdown won't display
+      Example content:
+
+      ```
+      [
+        { id: 1, label: "foo.bar" },
+        "separator",
+        { id: 2, translatedLabel: "FooBar" },
+        { id: 3 label: "foo.baz", icon: "times" }
+      ]
+      ```
+
+    - options: accepts a hash of optional attributes
+      - smallHeader: true/false, if true "btn-small" class will be added to header.
```

Comment:
A bool for adding a class doesn't scale well if you'd ever want to add another class to that header/button. Wouldn't explicitly passing a class list be better?

Reply:
yes agreed it feels a little bit hacky... I just thought this is a very common case, so I could make it explicit as `btn-small` requires you to know the internals of the dropdown